controllers/external-accounts.js

```diff
@@ -119,4 +120,94 @@ const syncExternalAccountData = async (req, res) => {
   }
 };
 
-module.exports = { addExternalAccountData, getExternalAccountData, syncExternalAccountData };
+/**
+ * Gets all group-roles
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const newSyncExternalAccountData = async (req, res) => {
+  try {
+    const [discordUserData, unArchivedRdsUsersData] = await Promise.all([
+      getDiscordMembers(),
+      fetchUsersForKeyValues("roles.archived", false),
+    ]);
+    let usersArchived = 0;
+    let usersUnArchived = 0;
+    let totalUsersProcessed = unArchivedRdsUsersData.length;
+
+    const discordUserIdSet = new Set();
+
+    discordUserData.forEach((discordUser) => discordUserIdSet.add(discordUser.user.id));
+
+    let updateUserList = [];
+
+    for (const rdsUser of unArchivedRdsUsersData) {
+      let userData = {};
+      if (discordUserIdSet.has(rdsUser?.discordId)) {
+        if (rdsUser.roles?.archived) usersUnArchived++;
+        userData = {
+          ...rdsUser,
+          roles: {
+            ...rdsUser.roles,
+            in_discord: true,
+            archived: false,
+          },
+        };
+
+        discordUserIdSet.delete(rdsUser.discordId);
+        updateUserList.push(userData);
+      } else {
+        usersArchived++;
+
+        userData = {
+          ...rdsUser,
+          roles: {
+            ...rdsUser.roles,
+            in_discord: false,
+            archived: true,
+          },
+        };
+        updateUserList.push(userData);
+      }
+    }
+    const updateUserBatchPromise1 = updateUsersInBatch(updateUserList);
+
+    const archivedUsersInDiscordList = await fetchUsersForKeyValues("discordId", [...discordUserIdSet]);
+    totalUsersProcessed += archivedUsersInDiscordList.length;
+    updateUserList = [];
+
+    for (const rdsUser of archivedUsersInDiscordList) {
+      usersUnArchived++;
+      const userData = {
+        ...rdsUser,
+        roles: {
+          ...rdsUser.roles,
+          in_discord: true,
+          archived: false,
+        },
+      };
+      updateUserList.push(userData);
+    }
+    const updateUserBatchPromise2 = updateUsersInBatch(updateUserList);
```

Comment:
- Should this be named `archieveUsersPromise` or something along those lines as we are archiving users here?

Reply:
Done